feat: untrusted fallback handler warning [SW-753]

tmjssz opened the pull request:
## What it solves

Resolves [SW-753](https://www.notion.so/safe-global/Show-a-warning-when-changing-the-fallback-handler-to-an-untrusted-contract-1828180fe57380e293e6e3b8adcb8f5d?pvs=4)

Show a warning in transactions that change the fallback handler to an untrusted contract.

## How this PR fixes it

### Enhancements to Fallback Handler Handling:

* [`apps/web/src/components/settings/FallbackHandler/index.tsx`](diffhunk://#diff-cf0fe2da0e00317b65d81009817517c4097468c6e23814760addc16c42ec1a01L1-L42): Replaced direct imports with new hooks `useCompatibilityFallbackHandlerDeployments`, `useIsOfficialFallbackHandler`, and `useIsTWAPFallbackHandler` to determine the type of fallback handler.

### Addition of Untrusted Fallback Handler Warnings:

* [`apps/web/src/components/transactions/Warning/index.tsx`](diffhunk://#diff-218cbf9f74c49a861f831eb9aeb2bd709ea1c22cf3300d8b1be5959063b1f303R10-R11): Added `UntrustedFallbackHandlerWarning` component to show warnings for untrusted fallback handlers in transaction details. [[1]](diffhunk://#diff-218cbf9f74c49a861f831eb9aeb2bd709ea1c22cf3300d8b1be5959063b1f303R10-R11) [[2]](diffhunk://#diff-218cbf9f74c49a861f831eb9aeb2bd709ea1c22cf3300d8b1be5959063b1f303R60-R84)
* [`apps/web/src/components/tx/confirmation-views/SettingsChange/UntrustedFallbackHandlerWarning.tsx`](diffhunk://#diff-2ce8205046ded1849dd027113b7c6c97d25775f039b60961a7fda3536a9ec720R1-R25): Created a new component to display an alert for untrusted fallback handlers and show it on the settings change confirmation view.

## How to test it

### Warning for queued transactions
1. Create a transaction that sets an untrusted fallback handler (e.g. `0xB4F1737Af37711e9A5890D9510c9bB60e170CB0D`). 
   [This script](https://github.com/safe-global/safe-wallet-monorepo/issues/4720#issuecomment-2567926655) can be used.
2. Open the transaction in the queue and observe a warning badge saying "Untrusted fallback handler".
3. Proceed to the transaction confirmation view and observe an alert for the fallback handler.

### Warning for executed transactions
1. Open an executed transaction that set an untrusted fallback handler, e.g. [this one](https://feat_untrusted_fallback_handler_warning--walletweb.review.5afe.dev/transactions/tx?safe=sep:0x12c0916493b9BEd2A9882d7722191660194eB47d&id=multisig_0x12c0916493b9BEd2A9882d7722191660194eB47d_0x68004c3de4755bca601f1b3e7b945c563a8b997f5a06dedc6063151989f8ee49)
2. Observe a warning badge saying "Untrusted fallback handler"

## Screenshots

### Queued transaction:
![Screenshot 2025-02-05 at 12 47 26](https://github.com/user-attachments/assets/d5285676-6231-4061-b6ef-985f2a5369d0) 
![Screenshot 2025-02-05 at 12 46 15](https://github.com/user-attachments/assets/84f59a47-caa9-4b3f-9e77-c50edc5b631d) 
![Screenshot 2025-02-05 at 12 48 45](https://github.com/user-attachments/assets/5169932a-ba37-4993-9d2d-b6b2a7c609be)

https://github.com/user-attachments/assets/82e20d83-16cf-4b95-8ac4-40f1e3b2b883

### Executed transaction:
![Screenshot 2025-02-05 at 12 48 03](https://github.com/user-attachments/assets/4df53563-94e6-47d3-b80c-b98f96c891e9)

## Checklist

- [ ] I've tested the branch on mobile 📱
- [ ] I've documented how it affects the analytics (if at all) 📊
- [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

katspaugh commented on apps/web/src/components/transactions/Warning/index.tsx:
```diff
@@ -55,6 +57,31 @@ export const DelegateCallWarning = ({ showWarning }: { showWarning: boolean }):
+  const isTWAPFallbackHandler = useIsTWAPFallbackHandler(fallbackHandler)
+
+  if (isOfficial || isTWAPFallbackHandler) {
+    return <></>
```
```suggestion
    return null
```

katspaugh commented on apps/web/src/components/transactions/Warning/index.tsx:
```diff
@@ -55,6 +57,31 @@ export const DelegateCallWarning = ({ showWarning }: { showWarning: boolean }):
+        <>
+          This transaction sets an <b>unofficial</b> fallback handler.
+          <br />
+          <b>Proceed with caution:</b> Ensure the fallback handler address is trusted and secure. If unsure, do not
```
```suggestion
          <b>Proceed with caution:</b> ensure the fallback handler address is trusted and secure. If unsure, do not
```

katspaugh commented on apps/web/src/components/tx/confirmation-views/SettingsChange/UntrustedFallbackHandlerWarning.tsx:
```diff
@@ -0,0 +1,25 @@
+    >
+      This transaction sets an <b>unofficial</b> fallback handler.
+      <br />
+      <b>Proceed with caution:</b> Ensure the fallback handler address is trusted and secure. If unsure, do not proceed.
```
```suggestion
      <b>Proceed with caution:</b> ensure the fallback handler address is trusted and secure. If unsure, do not proceed.
```

katspaugh commented on apps/web/src/hooks/useCompatibilityFallbackHandlerDeployments.ts:
```diff
@@ -0,0 +1,18 @@
+  const chain = useCurrentChain()
+
+  return useMemo(() => {
+    if (!chain || !safe.version) return undefined
+    return getCompatibilityFallbackHandlerDeployments({ network: chain?.chainId, version: safe.version })
+  }, [safe.version, chain])
```
```suggestion
    if (!chain?.chainId || !safe.version) return undefined
    return getCompatibilityFallbackHandlerDeployments({ network: chain?.chainId, version: safe.version })
  }, [safe.version, chain?.chainId])
```

katspaugh reviewed (approved):
Nice work!

The warning text is repeated in two components, consider extracting it into a shared component or variable.

tmjssz commented:
@katspaugh Thanks!

I updated the PR with all the changes you suggested. Also, I extracted the warning message into a new `UntrustedFallbackHandlerTxText` component.

liliya-soroka commented:
no warning if the sett unofficial fallbackhandler is one of a few actions in the tx

liliya-soroka commented:
Verified.
The warning in case of a few actions where one of them is set unofficial fallbackhandler will be moved to a new task

tmjssz commented:
> Verified. The warning in case of a few actions where one of them is set unofficial fallbackhandler will be moved to a new task

This is the follow-up task for multisend transactions: #4926
